internal/runtime/internal/worker/worker_pool.go

```diff
@@ -4,16 +4,19 @@ import (
 	"fmt"
 	"runtime"
 	"sync"
+	"time"
 )
 
 type Pool interface {
 	// Stop stops the worker pool. It does not wait to drain any internal queues, but it does wait for the currently
-	// running tasks to complete. It must only be called once.
-	Stop()
+	// running tasks to complete up to the specified timeout. It must only be called once.
+	// Returns an error if the timeout is exceeded before all workers have stopped, indicating tasks that
+	// take longer than timeout to complete.
+	Stop(timeout time.Duration) error
```

Comment:
Maybe split out this part into another pr? Then it won't be blocked by Prometheus changes. It would also make this change easier to track not being bundled up with "unrelated" changes

Reply:
Ok, was hoping it's not too much to review. Will do separate PR.